src/Avalonia.Controls/TabControl.cs

```diff
@@ -306,5 +309,17 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs chang
                     change.GetNewValue<IInputElement?>());
             }
         }
+
+        private void UpdateTabStripPlacement()
+        {
+            for (int i = 0; i < Items.Count; i++)
+            {
+                var control = ContainerFromIndex(i);
+                if (control is TabItem tabItem)
+                {
+                    tabItem.TabStripPlacement = TabStripPlacement;
+                }
+            }
+        }
```

Comment:
You can iterate over ItemsPresenterPart children

Reply:
Done